### DISCUSSION TITLE
SQL LIKE query

### DISCUSSION THREAD

**Discussion by AndiLeni** [msiemens/tinydb#363] [Q&A] [answered]
Hello,
is there an equivalent command to SQLs LIKE command?
F.e. to search for an entry with a field x and I only know a part of this string.

Kind regards
Andreas

**Comment by msiemens** [comment, marked as answer]
Hey @AndiLeni, you could use the `matches` and `search` search operators to search for a regular expression:

```python
# Full item has to match the regex:
db.search(User.name.matches('[aZ]*'))

# Any part of the item has to match the regex:
db.search(User.name.search('b+'))
```

See https://tinydb.readthedocs.io/en/latest/usage.html#advanced-queries for details on the search operation and https://docs.python.org/3/library/re.html#regular-expression-syntax on the regular expression syntax.

**Reply by AndiLeni** [reply]
Thanks, this is exactly what I was looking for!
